src/testshade/rs_simplerend.cpp

```diff
@@ -184,3 +185,26 @@ rs_transform_points(OSL::ShaderGlobals* /*sg*/, OSL::StringParam /*from*/,
 {
     return false;
 }
+
+OSL_RSOP bool
+rs_get_attribute_fallback(void* _sg, const char* _object, const char* _name, long long _type, int derivatives, int index, void* result)
+{
+    OSL::ShaderGlobals* sg = reinterpret_cast<OSL::ShaderGlobals*>(_sg);
+    const OSL::StringParam object = OSL::bitcast<OSL::ustringrep>(_object);
+    const OSL::StringParam name = OSL::bitcast<OSL::ustringrep>(_name);
+    const OSL::TypeDesc type = OSL::bitcast<OSL::TypeDesc>(_type);
+
+    return sg->renderer->get_array_attribute(sg, derivatives, object, type, name, index, result);
```

Comment:
This would work on the host, but not on a device because RenderServices isn't valid.  However RenderState is valid, it should exist on the device side.
What really needs to happen is to have utilize the RenderState provide rs_get_attribute_* for all the possibilities or use and int/enum and switch to access the correct data member (perhaps relying on compiler optimization).  If possibile, explicit functions would be better (not require as much work by optimizer).

IE:

```
OSL_RSOP bool
rs_get_attribute_camera_resolution(void* _sg, void *val)
{
       RenderState* rs = reinterpret_cast<RenderState*>(sg->renderstate);
        ((int*)val)[0] = rs->xres;
        ((int*)val)[1] = rs->yres;
        return true;
}
```

and the spaghetti code or map lookup would happen inside the` SimpleRenderer::build_attribute_getter`
and more bonus points if you can use the correct type for the result value, IE:
`rs_get_attribute__camera_resolution(void* _sg, int val[2])`
which would require not casting the pointer to void * on the llvm JIT side.